plugin/src/main/java/io/fabric8/maven/plugin/BuildMojo.java

```diff
@@ -241,7 +293,191 @@ private void executeOpenShiftBuild(ServiceHub hub, ImageConfiguration imageConfi
+
+        waitForOpenShiftBuildToComplete(client, buildName, build);
+
+        updateImageStreamTags(client, imageConfig, buildName, build);
+    }
+
+
+    private void waitForOpenShiftBuildToComplete(OpenShiftClient client, String buildConfigName, Build build) {
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        String buildName = getName(build);
+        Watcher<Build> buildWatcher = new Watcher<Build>() {
+            @Override
+            public void eventReceived(Action action, Build resource) {
+                if (isBuildCompleted(action, resource)) {
+                    latch.countDown();
+                }
+            }
+
+            @Override
+            public void onClose(KubernetesClientException cause) {
+            }
+        };
+        log.info("Waiting for build " + buildName + " to complete...");
+        try (Watch watcher = client.builds().withName(buildName).watch(buildWatcher)) {
+            while (latch.getCount() > 0L) {
+                try {
+                    latch.await();
+                } catch (InterruptedException e) {
+                    // ignore
+                }
+            }
+            log.info("Build " + buildName + " completed!");
+        }
+    }
+
+
+    /**
+     * Lets update the ImageStream to include the correct tags
+     */
+    private void updateImageStreamTags(OpenShiftClient client, ImageConfiguration imageConfig, String buildConfigName, Build build) throws MojoExecutionException {
+
+        try {
+            File manifest = openshiftManifest;
+            if (!Files.isFile(manifest)) {
+                throw new MojoFailureException("No such generated manifest file: " + manifest);
+            }
+
+            String namespace = clusterAccess.getNamespace();
+            Controller controller = new Controller(client);
+
+            boolean updated = false;
+            Set<HasMetadata> entities = loadResources(client, controller, namespace, manifest, project, log);
+            for (HasMetadata entity : entities) {
+                if (entity instanceof ImageStream) {
+                    ImageStream is = (ImageStream) entity;
+                    String imageStreamName = KubernetesHelper.getName(is);
+                    if (Objects.equals(buildConfigName, imageStreamName)) {
+                        if (updateImageStreamTag(client, imageConfig, is, buildConfigName)) {
+                            updated = true;
+                        }
+                    }
+                }
+            }
+            if (updated) {
+                // lets store the entities again!
+                KubernetesList entity = new KubernetesListBuilder().withItems(new ArrayList<>(entities)).build();
+                File resourceFileBase = new File(this.targetDir, ResourceClassifier.OPENSHIFT.getValue());
+                AbstractResourceMojo.writeResourcesIndividualAndComposite(entity, resourceFileBase, ResourceFileType.yaml, log);
```

Comment:
I am bit ambivalent here, since the `BuildMojo` actually does *not* write resource descriptors, that's the task of the `DeployMojo`. Yes, it does create `BuildConfig` and `ImageStream` objects but only live for doing the build (not as part of the deployment). 

Not sure whether this can be kept this way, however I would really love to leave the writing of resources to disk to the `ResourceMojo` (separation of concerns). If its only for the watch functionality, isn't a live update sufficient ?